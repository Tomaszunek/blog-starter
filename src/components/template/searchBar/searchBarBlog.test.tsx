import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import SearchBarBlog, { AricleTypes, SearchContentType} from './searchBarBlog'

describe('Search Bar', () => {
  describe('Search Bar should render', () => {
    let app:any;
    beforeEach(() => {
      app = mount(<SearchBarBlog searchContentType={SearchContentType.article} articleType={AricleTypes.motivation} />);
    })
    it('render correctly date component', () => {
      const tree = renderer
        .create(<SearchBarBlog searchContentType={SearchContentType.article} articleType={AricleTypes.motivation} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should receive content type (article or products)', () => {
      expect(app.find('.contentType').instance().value).toEqual("article");
    });    
    it('should have categories in form', () => {
      expect(app.find('.categoryCont').exists()).toBe(true);
    });
    it('should have form', () => {
      expect(app.find('form').exists()).toBe(true);
    });
    it('should have submit button', () => {
      expect(app.find('button').exists()).toBe(true);
    }); 
  });

  describe('Search Bar function should correct works', () => {    
    it('should handle change content type select', () => {
      // 
    // it('should handle change main type', () => {})
    // it('should handle change article type', () => {})     
    // it('should handle change categories select', () => {})    
    // it('should return correct json on submit', () => {});
  });
});