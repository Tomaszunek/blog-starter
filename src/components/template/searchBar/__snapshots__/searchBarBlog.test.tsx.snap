// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Search Bar should render render correctly date component 1`] = `
<div
  className="searchBar"
>
  <form
    action=""
  >
    <select
      className="contentType"
      defaultValue={
        Array [
          "article",
        ]
      }
      multiple={true}
      name="findType"
      onChange={[Function]}
    >
      <option
        value="article"
      >
        article
      </option>
      <option
        value="product"
      >
        product
      </option>
    </select>
    <div
      className="categoryCont"
    >
      Categories
    </div>
    <button>
      Find
      article
    </button>
  </form>
</div>
`;
